Rack/app/models/booking.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Booking < ActiveRecord::Base
+  belongs_to :customer
+  belongs_to :sup
+  has_and_belongs_to_many :gears
+
+  validates :start_date, :end_date, presence: true
+  validate :end_date_after_start_date
+
+  def booked_date_range
+    { from: :start_date, to: :end_date }
+  end
+
+  private
+
+  def end_date_after_start_date
+    errors.add(:end_date, 'must be after the start date') unless end_date < start_date
```

Comment:
you are adding error here when `end_date` is later than `start_date`

Reply:
not a mandatory method, added simply, for filling.

Reply:
I agree with you, but it has to work correctly. The best way here it's using `if` instead of `unless`

Reply:
why Rubocop writes : IF to long use UNLESS  bla bla bla?

Reply:
Rubocop wants that you should use `if` expression after `errors.add`, without wrapping around this action